### DISCUSSION TITLE
Get option key to act as meta/Esc+ on MacOS

### DISCUSSION THREAD

**Discussion by wcauchois** [kovidgoyal/kitty#4188] [Q&A] [answered]
Hello! I'm just trying out Kitty on MacOS and I'm finding that some of the emacs-esque shortcuts related to the alt/option key aren't working. Namely alt+b and alt+f to go backwards and forwards a word (equivalently alt+left and alt+right).

I currently use iTerm2, and from what I recall I did have to do something to iTerm to get this working, I _think_ it's related to this configuration to change left option to send "Esc+" (which I think is "Meta"? I'm not too familiar with these terms, just trying to get my muscle-memory shortcuts working!)

![image](https://user-images.githubusercontent.com/300544/140168870-74b326cc-248d-4a81-85b6-546d5de9cd4d.png)

I did some googling and I found issue #123, but even though it's closed I couldn't figure out what exactly the resolution was. At some point there's a link to StackOverflow and people suggest installing some other tool that changes shortcuts system-wide, but this doesn't seem that appealing to me. Is there a way to do this inside Kitty, or am I missing something from that ticket?

Thanks so much for any help!

**Comment by Luflosi** [comment, marked as answer]
See https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.macos_option_as_alt.

**Reply by wcauchois** [reply]
Amazing that worked perfectly. Thank you!

**Reply by mok0** [reply]
That is not a solution if you want to use Emacs in the terminal, which for example is relevant if you contact a remote host.

WezTerm solves this problem by treating the left Option key as the Alt modifier with no composition effects, while the right Option key performs composition.

**Reply by kovidgoyal** [reply]
A solution that WezTerm copied from kitty. If you had bothered to read you would see that macos_option_as_left can take the values, left, right, yes and both.

**Reply by mok0** [reply]
And that you copied from iTerm? Anyway thank you for replying. I did miss that information when reading the documentation. However, I have now configured Wezterm as I like it, plus it has a couple of other features I prefer over kitty, including a friendly and polite developer.

**Reply by miranquil** [reply]
This is not perfectly same as it does in iTerm2 or other shells. 
For instance, input `cd ~/.config/kitty`, and press `opt+b` after related configuration. 
On iTerm2(and Termius. etc), the cursor will stop before `k` (step by word as expected). But on kitty, the cursor will stop before `~`. Seems kitty takes the whole sentence as a word.

**Comment by kovidgoyal** [comment]
On Thu, Sep 21, 2023 at 05:52:32AM -0700, Morten Kjeldgaard wrote:
> And that you copied from iTerm? Anyway thank you for replying. I did miss that information when reading the documentation. However, I have not configured Wezterm as I like it, plus it has a couple of other features I prefer over kitty, including a friendly and polite developer.

Sadly, it has rude and entitled users who think it's OK to come to other
projects pages and post about competing products without even bothering
to do a modicum of homework. WezTerm is *most welcome* to you.
